org.sonarlint.eclipse.ui/src/org/sonarlint/eclipse/ui/internal/backend/SonarLintEclipseClient.java

```diff
@@ -254,20 +261,55 @@ public void showSmartNotification(ShowSmartNotificationParams params) {
 
     Display.getDefault().asyncExec(() -> new DeveloperNotificationPopup(connectionOpt.get(), params, connectionOpt.get().isSonarCloud()).open());
   }
-
-  // project synchronizations not yet set to be handled by sonarlint-core
+  
+  /** Start IDE progress bar for backend jobs running out of IDE scope */
   @Override
   public CompletableFuture<Void> startProgress(StartProgressParams params) {
-    return null;
+    return BackendProgressJobScheduler.get().startProgress(params);
   }
-
-  // project synchronizations not yet set to be handled by sonarlint-core
+  
+  /** Update / finish IDE progress bar for backend jobs running out of IDE scope */
   @Override
   public void reportProgress(ReportProgressParams params) {
+    if (params.getNotification().isLeft()) {
+      BackendProgressJobScheduler.get().update(params.getTaskId(), params.getNotification().getLeft());
+    } else {
+      BackendProgressJobScheduler.get().complete(params.getTaskId());
+    }
   }
 
-  // project synchronizations not yet set to be handled by sonarlint-core
+  /**
+   *  After synchronization of the backend, the taint vulnerabilities view should be updated with the remote findings
+   *  and also an analysis of the opened files should be triggered.
+   */
   @Override
   public void didSynchronizeConfigurationScopes(DidSynchronizeConfigurationScopeParams params) {
+    params.getConfigurationScopeIds().stream()
+      .map(id -> resolveProject(id))
+      .filter(Optional::isPresent)
+      .map(Optional::get)
+      .collect(Collectors.toSet())
+      .forEach(project -> {
+      var openedFiles = PlatformUtils.collectOpenedFiles(project, f -> true);
+      if (!openedFiles.isEmpty() && openedFiles.containsKey(project) && !openedFiles.get(project).isEmpty()) {
```

Comment:
Partly, as it checks the editor files it can be possible that either no file is opened or files but not from the project. The third check was removed by me now.